qiita_db/metadata_template/test/test_prep_template.py

```diff
@@ -1340,11 +1340,26 @@ def test_artifact_setter_error(self):
 
     def test_artifact_setter(self):
         pt = qdb.metadata_template.prep_template.PrepTemplate.create(
-            self.metadata, self.test_study, self.data_type_id)
+            self.metadata, self.test_study, '16S')
         self.assertEqual(pt.artifact, None)
         artifact = qdb.artifact.Artifact.create(
             self.filepaths, "FASTQ", prep_template=pt)
         self.assertEqual(pt.artifact, artifact)
+
+        # here we can test that we can properly create a workflow
```

Comment:
This seems like a limited test given the complexity of the function, are there edge cases beyond the `ValueError` check below that should be examined?

Reply:
Good point, I'll add more tests.

Reply:
Thanks!

Reply:
Just FYI, I checked everything that came to mind and I think those "simple" tests actually cover all scenarios by having to create a full pipeline and then checking that full pipeline and not having anything else to create.